### PR TITLE
Account Recovery Step 2

### PR DESCRIPTION
**Pull request by kimlisa**
part of RFD 0029: https://github.com/gravitational/teleport/blob/master/rfd/0029-account-lifecycle.md
split from larger PR for easier review: https://github.com/gravitational/teleport/pull/7405

### Description
- Implemented `ApproveAccountRecovery`, which is step 2 in the account recovery process. It creates a token that allows user to perform protected actions without being logged in:
  - User provides their password if recovering second factor, or provides second factor if recovering password
  - Maxing out recovery attempt locks the user from logging in, resets the recovery attempts, and deletes the token which will force the user to restart the recovery flow from step 1 (StartAccountRecovery). The recovery attempt for this step is a combined total attempt with step 1 (StartAccountRecovery).
  - Upon valid authn credential, creates a recovery approved token that expires in  15 minutes
- Placed rate limit

### CODE REVIEW

**Review by codingllama — COMMENTED** [review]
Nicely done.

File: lib/auth/accountrecovery_test.go [hunk 1]
```diff
@@ -317,6 +315,218 @@ func TestStartAccountRecovery_UserErrors(t *testing.T) {
...
+					AuthnCred: &proto.ApproveAccountRecoveryRequest_Password{Password: u.password},
+				}
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			// Acquire a start token.
+			startToken, err := srv.Auth().createRecoveryToken(ctx, u.username, UserTokenTypeRecoveryStart, c.recoverType)
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply lib/auth/accountrecovery_test.go:405]: I could but would require more boilerplate, especially with how different recover type will require different authentication to get the start token.

File: lib/auth/accountrecovery_test.go [hunk 2]
```diff
@@ -360,76 +570,39 @@ func createUserWithSecondFactorAndRecoveryCodes(srv *TestTLSServer, secondFactor
...
 
-	otpCode, err := totp.GenerateCode(secrets.GetOTPKey(), srv.Clock().Now())
+	// Insert a totp device manually.
+	otpSecret := base32.StdEncoding.EncodeToString([]byte("def456"))
+	dev, err := services.NewTOTPDevice("otp", otpSecret, srv.Auth().clock.Now())
 	if err != nil {
-		return "", "", trace.Wrap(err)
+		return nil, trace.Wrap(err)
+	}
+	if err = srv.Auth().UpsertMFADevice(ctx, username, dev); err != nil {
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply lib/auth/accountrecovery_test.go:595]: i also stole this from another test somewhere (:

**Review by codingllama — APPROVED** [review]
LGTM.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**